packages/react-native/Libraries/AppDelegate/RCTRootViewFactory.h

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#import <React/RCTUtils.h>
+#import <React/RCTBridge.h>
+#import <React/RCTRootView.h>
+
+@protocol RCTCxxBridgeDelegate;
+@protocol RCTComponentViewFactoryComponentProvider;
+@class RCTBridge;
+@class RCTRootView;
+@class RCTSurfacePresenterBridgeAdapter;
+
+NS_ASSUME_NONNULL_BEGIN
+
+
+#pragma mark - Blocks' definitions
+typedef UIView*_Nonnull (^RCTCreateRootViewWithBridgeBlock) (RCTBridge *bridge, NSString *moduleName, NSDictionary *initProps);
+typedef RCTBridge*_Nonnull (^RCTCreateBridgeWithDelegateBlock) (id<RCTBridgeDelegate> delegate, NSDictionary* launchOptions);
+
+
+#pragma mark - RCTRootViewFactory Configuration
+@interface RCTRootViewFactoryConfiguration : NSObject
+
+/// This property controls whether the App will use the Fabric renderer of the New Architecture or not.
+@property(nonatomic, assign) BOOL fabricEnabled;
+
+/// This property controls whether React Native's new initialization layer is enabled.
+@property(nonatomic, assign) BOOL bridgelessEnabled;
+
+/// This method controls whether the `turboModules` feature of the New Architecture is turned on or off
+@property(nonatomic, assign) BOOL turboModuleEnabled;
+
+/// Return the bundle URL for the main bundle.
+@property(nonatomic) NSURL *bundleURL;
+
+/**
+ * Use this method to initialize a new instance of `RCTRootViewFactoryConfiguration` by passing a `bundleURL`
+ *
+ * Which is the location of the JavaScript source file. When running from the packager
+ * this should be an absolute URL, e.g. `http://localhost:8081/index.ios.bundle`.
+ * When running from a locally bundled JS file, this should be a `file://` url
+ * pointing to a path inside the app resources, e.g. `file://.../main.jsbundle`.
+ *
+ */
+- (instancetype) initWithBundleURL:(NSURL *)bundleURL;
```

Comment:
what I don't like particularly about this initializer is that it makes the props mutable.

I'd rather have:
- readonly properties
- a static factory method that gets all the information it needs:
```objc
+ (RCTRootViewFactoryConfiguration)rootViewFactoryConfigurationWith:(NSURL *)bundleURL
                                                     newArchEnabled:(BOOL)newArchEnabled
                                                 turboModuleEnabled:(BOOL)turboModuleEnabled
                                                 ...
                                                 ;

```

Reply:
I've changed it to be the initializer and made properties readonly